code/modules/mob/living/carbon/xenomorph/castes/spiderling/spiderling.dm

```diff
@@ -30,6 +30,8 @@
 /datum/ai_behavior/spiderling
 	target_distance = 1
 	base_action = ESCORTING_ATOM
+	// is the spiderling currently resting?
+	var/spiderling_is_resting
```

Comment:
HAS_TRAIT_FROM_ONLY(src, TRAIT_FLOORED, RESTING_TRAIT)
?

Reply:
I didnt use HAS_TRAIT_FROM_ONLY but maybe its ok?

Reply:
Has a higher chance of breaking but whatever